Mainframe3270/x3270.py

```diff
@@ -81,51 +81,44 @@ def open_connection(
             | Open Connection | Hostname |
             | Open Connection | Hostname | LU=LUname |
             | Open Connection | Hostname | port=992 |
-            | ${extra_args}   | Create List | -accepthostname | myhost.com | -cafile | ${CURDIR}/cafile.crt |
-            } append to list  | ${extra_args} | -port | 992 |
+            | @{extra_args}   | Create List | -accepthostname | myhost.com | -cafile | ${CURDIR}/cafile.crt |
+            | Append To List  | ${extra_args} | -port | 992 |
             | Open Connection | Hostname | extra_args=${extra_args} |
             | Open Connection | Hostname | extra_args=${CURDIR}/argfile.txt |
         """
-        self.host = host
-        self.lu = LU
-        self.port = port
-        if extra_args:
-            extra_args = self._process_args(extra_args)
-            argport = ""
```

Comment:
Instead of extracting the port from the extra_args, I implemented the _port_in_extra_args() function.
If the port is specified in the extra_args, we can connect just using the hostname.

I was thinking, if a user goes to specify the port via extra_args, it is probably a conscious choice to do so, and no warning or notification is needed.

But there is still something I do not like. As the name suggests, extra_args is supposed to be a set of arguments to the Emulator in addition to the parameters of Open Connection. Since port is already a parameter why specify it in extra_args?

Reply:
I agree. But when yo do not use port and want to specify it in extra_args in my test still port 23 got used as it is default.
I am not that experienced in python that I know the correct way on how to discern a set variable or a defaulted one. Do you know?